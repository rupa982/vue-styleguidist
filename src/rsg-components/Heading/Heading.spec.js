import React from 'react';
import { HeadingRenderer } from './HeadingRenderer';

it('renderer should render H tag', () => {
	const actual = shallow(
		<HeadingRenderer classes={{}} level={1} slug="heading">Heading</HeadingRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('should compose passed class names', () => {
	const actual = shallow(
		<HeadingRenderer
			classes={{ heading: 'baseHeadingClass' }}
			className="customClass"
			level={1}
			slug="test"
		>
			Test heading
		</HeadingRenderer>
	);

	expect(actual.find('h1').prop('className')).toBe('baseHeadingClass customClass');
});

[1, 2, 3, 4, 5, 6].forEach(level => {
	it(`should render h${level} tag with anchor link inside`, () => {
		const actual = shallow(
			<HeadingRenderer classes={{}} level={level} slug={`h${level}`}>{`H${level}`}</HeadingRenderer>
		);
		expect(actual).toMatchSnapshot();
	});
});
