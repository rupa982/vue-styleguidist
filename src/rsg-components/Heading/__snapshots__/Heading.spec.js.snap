// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render H tag 1`] = `
<h1
  className=""
  id="heading"
>
  <a
    aria-hidden={true}
    href="#heading"
  >
    #
  </a>
  Heading
</h1>
`;

exports[`should render h1 tag with anchor link inside 1`] = `
<h1
  className=""
  id="h1"
>
  <a
    aria-hidden={true}
    href="#h1"
  >
    #
  </a>
  H1
</h1>
`;

exports[`should render h2 tag with anchor link inside 1`] = `
<h2
  className=""
  id="h2"
>
  <a
    aria-hidden={true}
    href="#h2"
  >
    #
  </a>
  H2
</h2>
`;

exports[`should render h3 tag with anchor link inside 1`] = `
<h3
  className=""
  id="h3"
>
  <a
    aria-hidden={true}
    href="#h3"
  >
    #
  </a>
  H3
</h3>
`;

exports[`should render h4 tag with anchor link inside 1`] = `
<h4
  className=""
  id="h4"
>
  <a
    aria-hidden={true}
    href="#h4"
  >
    #
  </a>
  H4
</h4>
`;

exports[`should render h5 tag with anchor link inside 1`] = `
<h5
  className=""
  id="h5"
>
  <a
    aria-hidden={true}
    href="#h5"
  >
    #
  </a>
  H5
</h5>
`;

exports[`should render h6 tag with anchor link inside 1`] = `
<h6
  className=""
  id="h6"
>
  <a
    aria-hidden={true}
    href="#h6"
  >
    #
  </a>
  H6
</h6>
`;
